tidy3d/components/source.py

```diff
@@ -1095,6 +1095,23 @@ def _bend_axis(self) -> Axis:
 """ Angled Field Sources one can use. """
 
 
+class AbstractAngularSpec(Tidy3dBaseModel, ABC):
+    """Abstract base for defining angular variability specifications for plane waves."""
+
+
+class FixedInPlaneKSpec(AbstractAngularSpec):
+    """Plane wave is injected such that its in-plane wavevector is constant. That is,
+    the injected field satisfies Bloch boundary conditions and its propagation direction is
+    frequency dependent.
+    """
+
+
+class FixedAngleSpec(AbstractAngularSpec):
+    """Plane wave is injected such that its propagation direction is frequency independent.
+    When using this option boundary conditions in tangential directions must be set to periodic.
+    """
```

Comment:
yes, there could be some additional parameters eventually after we learn more about performance of this in various situations, but not at the present time. I wonder, if it would be better to follow the field `subpixel` role. Have it initially as bool (`fixed_angle=True`), and later generalize to a spec if needed (`fixed_angle=FixedAngleSpec(...)`). I don't think we will need to add any parameters for regular (bloch) plane waves